builtin/credential/jwt/provider_kubernetes.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2025 OpenBao a Series of LF Projects, LLC
+// SPDX-License-Identifier: MPL-2.0
+
+package jwtauth
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/hashicorp/go-cleanhttp"
+)
+
+// Global variables instead of const to allow test cases to overwrite paths.
+var (
+	// localJWTPath is the path to the Kubernetes Service Account JWT token file.
+	localJWTPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Wonder if we can pull this out of a k8s library somewhere? I suppose by definition, we know we're on Linux 😆

Reply:
I don’t think so, even Kubernetes’ client-go has the default path hardcoded and not exposed publicly: https://github.com/kubernetes/client-go/blob/a296bd743c2dd954f56664f6ee7bc8558f7d8e32/rest/config.go#L544-L547